bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -314,11 +369,50 @@ func (p *packer) encodeTuple(t Tuple, nested bool) {
 // call Pack when using a Tuple with a FoundationDB API function that requires a
 // key.
 func (t Tuple) Pack() []byte {
-	p := packer{buf: make([]byte, 0, 64)}
+	p := newPacker()
 	p.encodeTuple(t, false)
 	return p.buf
 }
 
+// PackWithVersionstamp packs the specified tuple into a key for versionstamp operations
+func (t Tuple) PackWithVersionstamp() ([]byte, error) {
+	hasVersionstamp, err := t.HasIncompleteVersionstamp()
+	if err != nil {
+		return nil, err
+	}
+
+	p := newPacker()
+	p.encodeTuple(t, false)
+
+	if hasVersionstamp {
+		var scratch [4]byte
+		binary.LittleEndian.PutUint32(scratch[:], uint32(p.versionstampPos))
+		p.putBytes(scratch[:])
+	}
+
+	return p.buf, nil
```

Comment:
The way the Java and Python bindings work, this will return an error if `hasVersiosntamp` is false. This is to protect the user from doing something like:

```java
tr.mutate(MutationType.SET_VERSIONSTAMPED_KEY, Tuple.from(UUID.randomUUID(), new byte[0], null, null).packWithVersionstamp(), new Tuple());
```

In this case, the first 10 bytes of the UUID will get overwritten with the versionstamp and the byte array and two nulls being removed from the tuple, which *probably* isn't what the user intended.

Reply:
👍